packages/plugin-assets-retry/src/runtime/asyncChunkRetry.ts

```diff
@@ -173,36 +196,29 @@ function ensureChunk(chunkId: string): Promise<unknown> {
       throw error;
     }
 
-    const context: AssetsRetryHookContext = {
-      times: existRetryTimes,
-      domain: nextDomain,
-      url: nextRetryUrl,
-      tagName: 'script',
-    };
-
+    // Start retry
     if (config.onRetry && typeof config.onRetry === 'function') {
       config.onRetry(context);
     }
 
     // biome-ignore lint/complexity/useArrowFunction: use function instead of () => {}
-    return new Promise(function (resolve) {
-      // TODO: options to set retryDelay
-      // setTimeout(() => {
-      resolve(ensureChunk(chunkId));
-      // }, 300);
-    })
-      .then((result) => {
-        if (typeof config.onSuccess === 'function') {
+    return ensureChunk(chunkId).then((result) => {
+      if (typeof config.onSuccess === 'function') {
+        const context = createAssetsRetryContext(
```

Comment:
Now there are only two context objects and `createAssetsRetryContext` seems not necessary.

Can we extend from the outside context object like this?

```js
const currentContext = {
  ...context,
  times: existRetryTimes
}
```

Reply:
okay